compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -539,25 +539,32 @@ execCompile inputFile outputFile opts (WriteInterface writeInterface) mbIfaceDir
         createDirectoryIfMissing True $ takeDirectory outputFile
         B.writeFile outputFile $ Archive.encodeArchive bs
 
-execLint :: FilePath -> Options -> Command
-execLint inputFile opts =
+execLint :: [FilePath] -> Options -> Command
+execLint inputFiles opts =
   Command Lint (Just projectOpts) effect
   where
      projectOpts = ProjectOpts Nothing (ProjectCheck "" False)
      effect =
        withProjectRoot' projectOpts $ \relativize ->
        do
          loggerH <- getLogger opts "lint"
-         inputFile <- toNormalizedFilePath' <$> relativize inputFile
-         opts <- setDlintDataDir opts
-         withDamlIdeState opts loggerH diagnosticsLogger $ \ide -> do
-             setFilesOfInterest ide (HashSet.singleton inputFile)
-             runActionSync ide $ getDlintIdeas inputFile
-             diags <- getDiagnostics ide
-             if null diags then
-               hPutStrLn stderr "No hints"
-             else
-               exitFailure
+         inputFiles <- getInputFiles inputFiles
+         forM_ inputFiles $ \inputFile -> do
+           inputFile <- toNormalizedFilePath' <$> relativize inputFile
+           opts <- setDlintDataDir opts
+           withDamlIdeState opts loggerH diagnosticsLogger $ \ide -> do
+               setFilesOfInterest ide (HashSet.singleton inputFile)
+               runActionSync ide $ getDlintIdeas inputFile
+               diags <- getDiagnostics ide
+               if null diags then
+                 hPutStrLn stderr "No hints"
+               else
+                 exitFailure
+     getInputFiles = \case
+       [] -> do
+           fs <- listFilesInside (pure . not . isPrefixOf ".daml" . takeFileName) "."
```

Comment:
this doesn’t seem quite right. We only want files in the source directory from your daml.yaml. We can probably share the logic that `daml build` uses to find those.

Reply:
using `getDamlRootFiles` now.